openupgradelib/openupgrade.py

```diff
@@ -896,15 +896,41 @@ def rename_xmlids(cr, xmlids_spec):
 
     :param xmlids_spec: a list of tuples (old module.xmlid, new module.xmlid).
     """
+    get_data_query = (
+        """SELECT res_id, model FROM ir_model_data
+        WHERE module=%s AND name=%s""")
+
     for (old, new) in xmlids_spec:
         if '.' not in old or '.' not in new:
             logger.error(
                 'Cannot rename XMLID %s to %s: need the module '
                 'reference to be specified in the IDs' % (old, new))
-        else:
-            query = ("UPDATE ir_model_data SET module = %s, name = %s "
-                     "WHERE module = %s and name = %s")
-            logged_query(cr, query, tuple(new.split('.') + old.split('.')))
+            continue
+
+        cr.execute(get_data_query, tuple(old.split('.')))
+        old_row = cr.fetchone()
+        if not old_row:
+            logger.warning('XMLID %s not found when renaming to %s', old, new)
+            continue
+
+        if old_row[1] == 'ir.module.category':
```

Comment:
Or, alternatively, add a parameter to the method to allow merging regardless of the model.

Reply:
Oh, that would be even better. But then what about the model and model_table?

Reply:
Determine table by replacing dots. Should work in 99.9% of the time.

Reply:
Maybe it would be better pass the table in the merge parameter you suggested. Imagine having xmlids you would like to merge and others that not, then you would have to create two lists and that would be a bit awful.

What do you think of the following?

```python
openupgrade.rename_xmlids(env.cr, [
    ('old_id1', 'new_id1'),  # no merge
    ('old_id2', 'new_id2', 'table2'), # merge
    ('old_id3', 'new_id3'), # no merge
    ...
])

def rename_xmlids(cr, xmlids_spec):
    for spec in xmlids_spec:
        if len(spec) == 3:
            old, new, table = spec
        else:
            old, new = spec
            table = None
        (...)
        if table:
            # proceed with the fetch new_row and merge code
        (...)
```

Reply:
Like I said, the current approach will work for 99.9% of the time, whereas the need to rename is only there for 0.01% of the time.

I pushed the new approach.

Reply:
Note that I took care to catch problems with the guessed table and a raise clear error message.